tests/jax/updating/terminators_test.py

```diff
@@ -0,0 +1,202 @@
+
+    def _set_stopped() -> None:
+        test_parameter_server.store.stopped = True
+
+    test_terminator = CountConditionTerminator(
+        config=MockCountConditionTerminatorConfig(
+            termination_condition=condition, termination_function=_set_stopped
+        )
+    )
+
+    for _ in range(15):
+        step_parameters(
+            test_parameter_server.store.parameters, list(condition.keys())[0]
+        )
+
+    test_terminator.on_parameter_server_run_loop_termination(test_parameter_server)
+
+    assert test_parameter_server.store.stopped is True
+
+
+@pytest.mark.parametrize("condition", count_condition_terminator_data())
+def test_count_condition_terminator_not_terminated(
+    condition: Dict, mock_parameter_server: SystemParameterServer
+) -> None:
+    """Test if count condition terminator does not terminate"""
+    test_parameter_server = mock_parameter_server
+
+    def _set_stopped() -> None:
+        test_parameter_server.store.stopped = True
+
+    test_terminator = CountConditionTerminator(
+        config=MockCountConditionTerminatorConfig(
+            termination_condition=condition, termination_function=_set_stopped
+        )
+    )
+
+    for _ in range(5):
+        step_parameters(
+            test_parameter_server.store.parameters, list(condition.keys())[0]
+        )
+
+    test_terminator.on_parameter_server_run_loop_termination(test_parameter_server)
+
+    assert test_parameter_server.store.stopped is False
+
+
+@pytest.mark.parametrize(
+    "fail_condition,failure", count_condition_terminator_failure_cases()
+)
+def test_count_condition_terminator_exceptions(
+    fail_condition: Dict,
+    failure: Type[Exception],
+    mock_parameter_server: SystemParameterServer,
+) -> None:
+    """Test count condition terminator exceptions"""
+
+    with pytest.raises(failure):
+        test_parameter_server = mock_parameter_server
+
+        def _set_stopped() -> None:
+            test_parameter_server.store.stopped = True
+
+        test_terminator = CountConditionTerminator(
+            config=MockCountConditionTerminatorConfig(
+                termination_condition=fail_condition, termination_function=_set_stopped
+            )
+        )
+
+        assert (
+            test_terminator.on_parameter_server_run_loop_termination(  # type:ignore
+                test_parameter_server
+            )
+            == failure
+        )
+
+
+def test_time_terminator_terminated(
+    mock_parameter_server: SystemParameterServer,
+) -> None:
+    """Test if time terminator terminates"""
+
+    test_parameter_server = mock_parameter_server
+
+    def _set_stopped() -> None:
+        test_parameter_server.store.stopped = True
+
+    test_terminator = TimeTerminator(
+        config=MockTimeTerminatorConfig(
+            run_seconds=0.5, termination_function=_set_stopped
+        )
+    )
+
+    test_terminator.on_parameter_server_init(test_parameter_server)
+
+    time.sleep(1)
```

Comment:
So you really don't want to use `time.sleep` in tests, it's considered bad practice. Rather set `run_second` to 0

Reply:
Good catch! Thank you. Changed it.